frontend/README.md

```diff
@@ -0,0 +1,48 @@
+# ARO-HCP-FRONTEND
+
+## Run the frontend
+
+```bash
+make frontend
+```
+
+To create a cluster, follow the instructions in [development-setup.md][../dev-infrastructure/docs/development-setup.md]
+
+## Available endpoints
+
+List the Operations for the Provider
+```bash
+curl -X GET "https://localhost:8443/providers/Microsoft.RedHatOpenshift/operations?api-version=2024-06-10-preview"
+```
+
+List HcpOpenShiftVersions Resources by Location
+```bash
+curl -X GET "https://localhost:8443/subscriptions/YOUR_SUBSCRIPTION_ID/locations/YOUR_LOCATION/providers/Microsoft.RedHatOpenshift/hcpOpenShiftVersions?api-version=2024-06-10-preview"
+```
+
+List HcpOpenShiftClusterResource Resources by Subscription ID
+```bash
+curl -X GET "https://localhost:8443/subscriptions/YOUR_SUBSCRIPTION_ID/providers/Microsoft.RedHatOpenshift/hcpOpenShiftClusters?api-version=2024-06-10-preview"
+```
+
+Get a HcpOpenShiftClusterResource
+```bash
+curl -X GET "https://localhost:8443/subscriptions/YOUR_SUBSCRIPTION_ID/resourceGroups/YOUR_RESOURCE_GROUP_NAME/providers/Microsoft.RedHatOpenshift/hcpOpenShiftClusters/YOUR_CLUSTER_NAME?api-version=2024-06-10-preview"
+```
+
+Create or Update a HcpOpenShiftClusterResource
+```bash
+curl -X PUT "https://localhost:8443/subscriptions/YOUR_SUBSCRIPTION_ID/resourceGroups/YOUR_RESOURCE_GROUP_NAME/providers/Microsoft.RedHatOpenshift/hcpOpenShiftClusters/YOUR_CLUSTER_NAME?api-version=2024-06-10-preview" -H "Content-Type: application/json" -d '{
```

Comment:
Maybe show referencing a filename for the cluster resource?  (`-d @mycluster.yaml`)  Probably would be the more common case than inlining data.

Reply:
Oh, also, curl offers a ``--json`` flag which could simplify the command a bit:

>        --json <data>
>               (HTTP) Sends the specified JSON data in a POST  request  to  the
>               HTTP  server.  --json  works  as a shortcut for passing on these
>               three options:
> 
>                --data [arg]
>                --header "Content-Type: application/json"
>                --header "Accept: application/json"
> 
>               There is no verification that the passed in data is actual  JSON
>               or that the syntax is correct.
> 
>               If  you  start  the data with the letter @, the rest should be a
>               file name to read the data from, or a single  dash  (-)  if  you
>               want  curl to read the data from stdin. Posting data from a file
>               named 'foobar' would thus be done with --json @foobar and to in‐
>               stead read the data from stdin, use --json @-.
> 
>               If  this option is used more than once on the same command line,
>               the additional data pieces will be concatenated to the  previous
>               before sending.
> 
>               The headers this option sets can be overridden with -H, --header
>               as usual.
>